src/routers/expenses.js

```diff
@@ -0,0 +1,127 @@
+'use strict';
+
+const express = require('express');
+const router = express.Router();
+const expensesService = require('../services/expenses');
+const userService = require('../services/user');
+
+router.use(express.json());
+
+router.get('/', (req, res) => {
+  const {
+    userId,
+    categories,
+    from,
+    to,
+  } = req.query;
+
+  const expenses = expensesService.getExpenses(userId, categories, from, to);
+
+  res.json(expenses);
+});
+
+router.post('/', (req, res) => {
+  const {
+    userId,
+    spentAt,
+    title,
+    amount,
+    category,
+    note,
+  } = req.body;
+
+  if (!(
+    userId !== undefined
+    && userService.getUserById(userId.toString())
+    && spentAt !== undefined
+    && title !== undefined
+    && amount !== undefined
+    && category !== undefined
+  )) {
+    res.sendStatus(400);
+
+    return;
+  }
```

Comment:
I would rather just implement checking with ! and || operators
```suggestion
   if (!userId || !user || !title || !amount || !category) {
    res.sendStatus(400);

    return;
  }
```

Reply:
that was my first thought, but values 0 and '' also could be there